### ISSUE TITLE
Write Unit Tests for `processing:core`

### ISSUE THREAD

**Issue by SableRaf** [processing/processing4#880]
The `processing:core` library is not changing a lot, so writing tests for it first makes the most sense.

_Full Description TBD_

**Comment by Rishab87** [comment]
Hi I would like to take upon this issue, can this be assigned to me?

**Comment by SableRaf** [comment]
Hi @Rishab87 and thanks for your interest in this!

Since you're new to the Processing codebase, I recommend starting by looking at the existing tests in processing:core ([here](https://github.com/processing/processing4/tree/main/core/test/processing)). This should give you a sense of the structure and approach used. Feel free to ask any clarifying questions before diving into implementation.

Since this issue is quite broad and doesn’t have a clear definition of done, a good approach would be to identify a specific area that would benefit from additional tests (such as a particular class or expanding an existing test class) and open a sub-issue for it.

Let me and @Stefterv know if you need any guidance. We’re happy to help!

**Comment by Rishab87** [comment]
Hi @SableRaf , I looked at the existing tests and went through few files roughly, I think writing tests for handling key events in PApplet.java can be added or we can expand current svg tests adding tests for other features, etc.

Honestly unit tests for most of the classes can be added ig but I think the most important ones should be done first which are critical for now do you have any specific tests in mind which are must have at this point?

**Comment by SableRaf** [comment]
Adding tests to handle key events sound like a good place to start!

Please make sure you keep PR(s) focused on one specific area of concern.

**Comment by Rishab87** [comment]
Sure @SableRaf , I'll keep that in mind? Should I create a new issue for it or just reference this one?

**Comment by SableRaf** [comment]
Thanks for your question! Please open a new issue with a well defined scope. This way it can be closed by your PR.

**Comment by Rishab87** [comment]
I've created a new issue and will soon open a PR for it https://github.com/processing/processing4/issues/965

**Comment by Vaivaswat2244** [comment]
Hello everyone, I am highly interested in writing tests for the classes.
Although recently many conributors are working on this. Can I get clarity on which classes should I get started on?
Then I can create issues for the new ones as well as @SableRaf mentioned.
